R/fileDependency.R

```diff
@@ -0,0 +1,26 @@
+#' Create a file dependency that can be accessed by the client using javascript.
+#' @export
+fileDependency <- function(filename, version = '0.0.1'){
+  htmltools::htmlDependency(
+    name = basename(tools:::file_path_sans_ext(filename)),
+    version = version,
+    src = dirname(filename),
+    attachment = basename(filename)
+  )
```

Comment:
Good points @jcheng5 . I did not think from this angle. I assumed the external file dependencies to be static resources, and this PR was an attempt to package the boilerplate code into something more automatic.

What you suggest here, would make this mechanism even more powerful, by allowing dynamic external dependencies. Your first comment seems like a simple enough fix that would give us reasonable mileage. However, if there is a way to insert the link directly in the page, I think it would work even better.